python/hpsmc/alignment/_parameter.py

```diff
@@ -0,0 +1,272 @@
+"""! representation of alignment parameters"""
+
+import re
+
+class Parameter :
+    """!
+    Representation of a single alignment parameter
+
+    This class also contains helpful functions for operating on sets of alignment
+    parameters e.g. parsing the map file or pede res file
+
+    Attributes
+    ----------
+    id : int
+        pede ID number as written in compact.xml, pede steering, and map files
+    name : str
+        human-readable name as written in map file
+    half : int
+        1 is for top and 2 is for bottom
+    trans_rot : int
+        1 is for translation and 2 is for rotation
+    direction : int
+        1 is for 'u', 2 is for v, and 3 is for w
+    mp_layer_id : int
+        "layer" ID number in millepede (i.e. axial and stereo sensors are separated)
+    val : float
+        value of parameter (if loaded from res file)
+    error : float
+        error of parameter (if loaded from res file)
+    active : bool
+        true if parameter is floating, false otherwise
+    """
+
+    idn_str_pattern = re.compile('^[12][12][123][0-9][0-9]$')
+
+    def __init__(self, idn, name, half, trans_rot, direction, mp_layer_id) :
+        self._id = int(idn)
+        self._name = name
+        self._half = int(half) # 1 or 2
+        self._trans_rot = int(trans_rot)
+        self._direction = int(direction)
+        self._mp_layer_id = int(mp_layer_id)
+
+        self._val = 0.0
+        self._error = -1.0
+        self._active = False
+
+    def float(self, yes = True) :
+        """!Set whether this parameter is floating/active or not"""
+        self._active = yes
+
+    def module(self) :
+        """!Get the module number from the millepede layer number
+
+        We group sensors in pairs to form modules and the layer number
+        for millepede counts up from the furthest upstream sensor. Thus,
+        we do integer-division by two to get the module number.
+        """
+        return self._mp_layer_id // 2
+
+    def layer(self) :
+        """!Get the human layer number
+
+        This is the module number but shifted by one since the first layer is
+        layer 1
+        """
+        return self.module() + 1
```

Comment:
Need to validate that this is how to get the module/layer number. I think this is how it is done in the front, but need to update so that it works for the back too.